### PR TITLE
Check in/37846/always set precheckin complete

### PR DESCRIPTION
**Pull request by acrollet**
## Description

We uncovered a couple of issues with the pre-check-in confirmations:

1. if the final confirmation page is skipped, the pre-check-in status is not set to complete
2. if any intermediate pages are skipped, the confirmations are incorrectly updated

This PR moves the pre-check-in call to the Complete page and adds a spinner while it is sent.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#37846

## Testing done

- local testing with mocks
- updated unit & e2e tests

## Screenshots

Screen Shot 2022-03-02 at 3 15 50 PM

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/check-in/hooks/useSessionStorage.jsx [hunk 1]
```diff
@@ -46,6 +46,20 @@ const useSessionStorage = (isPreCheckIn = true, maxValidateAttempts = 3) => {
     [SESSION_STORAGE_KEYS],
   );
 
+  const setComplete = useCallback(
+    (window, complete) => {
+      setSessionKey(window, SESSION_STORAGE_KEYS.COMPLETE, { complete });
```
Comment (mdewey) [inline src/applications/check-in/hooks/useSessionStorage.jsx:51]: This is now the second (at least) piece of data we are storing in session, should we start to think of session structuce?

**Review by mdewey — COMMENTED** [review]
Minor technical conversations, but nothing crazy. Good Work!

File: src/applications/check-in/api/local-mock-api/e2e/ApiInitializer.js [hunk 2]
```diff
@@ -190,12 +190,12 @@ class ApiInitializer {
           extraValidation(req);
         }
         req.reply(preCheckInData.post.createMockSuccessResponse('some-token'));
-      });
+      }).as('preCheckin');
```
[the remark this answers is not in the document]
Comment (acrollet) [reply src/applications/check-in/api/local-mock-api/e2e/ApiInitializer.js:193]: good call, I'll update :-)

[the remark this answers is not in the document]
Comment (acrollet) [reply src/applications/check-in/api/local-mock-api/e2e/ApiInitializer.js]: @mdewey I decided to err on the side of being super descriptive, let me know what you think.

File: src/applications/check-in/hooks/useSessionStorage.jsx (hunk 1 above)
Comment (acrollet) [reply src/applications/check-in/hooks/useSessionStorage.jsx:51]: let's chat about it after ToT today if you're free :-)

**Review by brianseek — APPROVED** [review]
There is one linting issue but other than that looks good!

**Comment by acrollet** [comment]
> There is one linting issue but other than that looks good!

fixed, thanks for the call-out!

**Pull request merged**